test/unit-tests/function/algebra/simplifyConstant.test.js

```diff
@@ -22,9 +22,47 @@ describe('simplifyConstant', function () {
     testSimplifyConstant('0.5 x', '1 / 2 x')
   })
 
-  it('should coalesce constants in a multi-argument expression', function () {
-    testSimplifyConstant('3 + x + 7 + y', '10 + x + y')
-    testSimplifyConstant('3 * x * 7 * y', '21 * x * y')
+  describe('should coalesce constants in a multi-argument expression', function () {
+    it('in the default context', function () {
+      testSimplifyConstant('3 + x + 7 + y', '10 + x + y')
+      testSimplifyConstant('3 * x * 7 * y', '21 * x * y')
+
+      // including an 'unevaluable' constant
+      // todo?: output here not currently as expected; instead outputting as '9 + "foo" + 11'.
+      // testSimplifyConstant('2 + 7 + "foo" + 3 + 8', '20 + "foo"')
+    })
```

Comment:
You might want to run this test with its current output since that is at least an improvement over what it produced *before* this PR, namely `9 + "foo" + 3 + 8`.